### PR TITLE
API: More elemwise and reduction functions

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi,

This PR adds another batch of elemwise and reduction functions.

### CODE REVIEW

File: tests/test_ops.py [hunk 1]
```diff
@@ -46,9 +46,22 @@ def my_custom_fun(arr1, arr2, arr3):
 
 
 @pytest.mark.parametrize(
-    "meth_name", ["__pos__", "__neg__", "__abs__"],
+    "meth_name", ["log", "log10", "log1p", "log2", "sqrt", "sign", "round", "exp", "expm1", "floor", "ceil"],
```
Comment (hameerabbasi) [inline tests/test_ops.py]: I'd test all the introduced methods

Comment (hameerabbasi) [reply tests/test_ops.py]: `meth_name` is no longer accurate as these are functions.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]

File: tests/test_ops.py (hunk 1 above)
Comment (mtsokol) [reply tests/test_ops.py]: Done! All new methods are tested and placed in `parametrize`.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
I'd also add checks for `dtype` and `shape`. Otherwise LGTM.

**Comment by mtsokol** [comment]
> I'd also add checks for `dtype` and `shape`. Otherwise LGTM.

Done! I added checks for these Tensor's attributes.

**Review by hameerabbasi — APPROVED** [review]

**Comment by willow-ahrens** [comment]
It occurs to me that perhaps the list of exported names in the module would be a good place to start counting supported functions, is that right?

**Comment by mtsokol** [comment]
We merged `array-api-tests` CI job in `sparse` (https://github.com/pydata/sparse/pull/668) that shows how many functions are already implemented (~46 right now).

**Pull request merged**